nvme-print.h

```diff
@@ -129,6 +129,33 @@ struct nvme_bar_cap {
 	__u8	rsvd61:3;
 };
 
+struct feature_fields {
+	union {
+		struct {
+			__u32 ab:3;
+			__u32 rsvd3:5;
+			__u32 lpw:8;
+			__u32 mpw:8;
+			__u32 hpw:8;
+		} arbitration;
+		struct {
+			__u32 ps:5;
+			__u32 wh:3;
+			__u32 rsvd8:24;
+		} power_management;
+		struct {
+			__u32 num:6;
+			__u32 rsvd6:26;
+		} lba_range_type;
+		struct {
+			__u32 tmpth:16;
+			__u32 tmpsel:4;
+			__u32 thsel:2;
+			__u32 rsvd22:10;
+		} temp_thresh;
```

Comment:
Bitfields should be avoided, as they are not endian safe. There are more reason for it but the big one is the endianess problem . 

It's better to work with shift and mask operations.

Reply:
Sorry probably this was mentioned before also. Before to fix the patch as mentioned to make sure let me check the endian problem behavior detail later.

Reply:
Still I could not prepare the big endian environment to check the endian problem behavior detail. I seems that the problem can be avoided to define the reserved data correctly as for example rsvd3, rsvd8, rsvd6 and rsvd22. Is this understanding not correct?

Reply:
For bitfielts it's necessary to define a big-endian and little-endian version (not to mention there is also another endianess but that's irrelevant these days 'middle-endian'). Anyway, just use SHIFT operators and this all goes away, because SHIFT operators a well defined.

Reply:
Thanks for your explanatoin. Just fixed the patch to use the shift operatiors.